### PR TITLE
fx: add google login middleware

### PR DESCRIPTION
**Pull request by musllim**
fx: add google login middleware

# Description
This app had no way to logging using google, the client needed this feature which will return the response to client.

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

# Type of change

Please check the options that are relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have created a loom video for the new feature or enhancement
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by ISHIMWE-Aime — DISMISSED** [review]
LGTM

**Comment by ISHIMWE-Aime** [comment]
But try to fix issues about what hound is tell you and circle ci please; 
despite from that you have done a great work.

**Review by SergeInGithub — COMMENTED** [review]
Please look into the error hound is picking and make the relevant changes.

**Review by emilenfc — APPROVED** [review]
LGTM

**Review by shyakadavis — APPROVED** [review]
LGTM

**Review by SergeInGithub — DISMISSED** [review]
LGTM

**Review by mucyoalexaxel — CHANGES_REQUESTED** [review]
@SergeInGithub Check suggested changes below.

File: src/controllers/googleAuth.controller.ts
[the remark this answers is not in the document]
Comment (musllim) [reply src/controllers/googleAuth.controller.ts:1]: verifyGoogleToken function will run the validation of credentials (google token sent in the body from Google) based on GOOGLE_CLIENT_ID and throw an error once on failure or return payload containing all credentials of Google user like email, given_name, family_name, picture.

**Review by mucyoalexaxel — APPROVED** [review]
LGTM

**Pull request merged**